src/ripple/app/misc/AmendmentTableImpl.cpp

```diff
@@ -108,220 +207,168 @@ class AmendmentTableImpl final : public AmendmentTable
     amendmentList_t getEnabled();
 
     // Amendments we support, do not veto, and are not enabled
-    amendmentList_t getDesired (enabledAmendments_t const&);
+    amendmentList_t getDesired (std::set <uint256> const&);
 };
 
-namespace detail
-{
-/** preEnabledAmendments is a static collection of amendments that are are
-   enabled at build time.
+//------------------------------------------------------------------------------
 
-   Add amendments to this collection at build time to enable them on this
-   server.
-*/
+AmendmentTableImpl::AmendmentTableImpl (
+        std::chrono::seconds majorityTime,
+        int majorityFraction,
+        Section const& supported,
+        Section const& enabled,
+        Section const& vetoed,
+        beast::Journal journal)
+    : m_lastUpdateSeq (0)
+    , m_majorityTime (majorityTime)
+    , mMajorityFraction (majorityFraction)
+    , m_journal (journal)
+{
+    assert (mMajorityFraction != 0);
 
-std::vector<AmendmentName> const preEnabledAmendments;
-}
+    std::lock_guard <std::mutex> sl (mutex_);
 
-void
-AmendmentTableImpl::addInitial (Section const& section)
-{
-    for (auto const& a : detail::preEnabledAmendments)
+    for (auto const& a : parseSection(supported))
     {
-        if (!a.valid ())
-        {
-            std::string const errorMsg =
-                (boost::format (
-                     "preEnabledAmendments contains an invalid hash (expected "
-                     "a hex number). Value was: %1%") %
-                 a.hexString ()).str ();
-            throw std::runtime_error (errorMsg);
-        }
-    }
+        auto s = add (a.first);
+
+        if (!a.second.empty ())
+            s->name = a.second;
```

Comment:
If we changed the param to `auto& a` we could `std::move (a.second)` here and save an allocation. OTOH, I'm less concerned about optimizations for on-off startup tasks like this. Your call.